misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt

```diff
@@ -157,21 +158,26 @@ internal class SessionFactoryService(
       value.typeParameters.setField("jsonColumnField", field)
     } else if (field.isAnnotationPresent(ProtoColumn::class.java)) {
       value.typeName = ProtoColumnType::class.java.name
-
       if (value.typeParameters == null) {
         value.typeParameters = Properties()
       }
       value.typeParameters.setField("protoColumnField", field)
     } else if (field.isAnnotationPresent(SecretColumn::class.java)) {
       value.typeName = SecretColumnType::class.java.name
-
       if (value.typeParameters == null) {
         value.typeParameters = Properties()
       }
       value.typeParameters.setProperty(SecretColumnType.FIELD_ENCRYPTION_KEY_NAME,
           field.getAnnotation(SecretColumn::class.java).keyName)
       value.typeParameters.setProperty(SecretColumnType.FIELD_ENCRYPTION_INDEXABLE,
           field.getAnnotation(SecretColumn::class.java).indexable.toString())
+    } else if (field.isAnnotationPresent(VerifiedColumn::class.java)) {
```

Comment:
That is a tricky question!
We've thought about it before, and it makes it does make sense and could be a nice feature to have.
On the other hand, it has 2 major disadvantages:
- It makes it hand to reason what should be the underlying datatype in the database. For example, if you want to have an encrypted/verified json, should it be defined as a `BLOB`? `VARBINARY`? `VARCHAR`?
- There's also the problem of the order of operations, how can we make sure for example that serialization happens before encryption/hashing, also in what form should it be cached?

In order to both avoid these issues and also provide explicitness to user defined entities, I've decided to not try and support that, *for now*.
If one wants to define a table that contains an encrypted or verified proto/json, one should take care of the serialization itself and provide a `String` to be hashed and persisted by the column annotated with `@VerifiedColumn`/`@SecretColumn`.

Reply:
I can totally see the reasoning on why it could be desirable to make these column attributes composable, but it's not a great fit with cryptographic best practices. In general, the best practice for modern encryption is to do authenticated encryption using an algorithm such as AES-GCM or ChaCha20Poly1305. So for our "Secret" columns, they are really already "Secret+Verified", but the 2nd part is implicit. I am pretty sure that we don't want to offer an option of making a Secret-but-not-verified column.

This might be something that we can make clearer with our attribute naming to clarify.